content/blog/2024-12-21-unikraft-releases-v0.18.0.mdx

```diff
@@ -0,0 +1,156 @@
+This is a long anticipated feature that allows efficient management of console devices, and enables the addition of new features, with the GDB stub being the first one to utilize this subsystem.
+
+### Further improvements of libukrandom
+
+_This feature was championed by [Michalis Pappas](https://github.com/michpappas) with great support from [Alex Apostolescu](https://github.com/Alex-deVis), [Maria Pana](https://github.com/mariapana), [Simon Kuenzer](https://github.com/skuenzer), and [Stefan Jumarea](https://github.com/StefanJum)._
+
+This release continues on the improvents introduced in Unikraft v0.17 for cryptographically secure randomness.
+`libukrandom` now includes a driver API, and the implementation of CPU-generated randomness has been reworked into conforming drivers.
+For more flexibility, the driver API introduces fallback operations: On x86_64, `RDSEED` can now fall back to `RDRAND`, addressing common issues faced by users running on legacy machines without `RDSEED` support, or those using TCG emulation on older versions of QEMU.
+
+In addition to the above, it is now possible to seed the CSPRNG via Unikraft's command-line, or through the device tree blob.
+
+Finally, the CSPRNG has been updated to now use 20 rounds (ChaCha20), and an implementation of the test vectors of [RFC8439](https://datatracker.ietf.org/doc/html/rfc8439) is available to execute via `libuktest`.
+
+For more info see the new documentation at [drivers/ukrandom/lcpu](https://github.com/unikraft/unikraft/tree/staging/drivers/ukrandom/lcpu/README.md) and [lib/ukrandom](https://github.com/unikraft/unikraft/tree/staging/lib/ukrandom/README.md).
+
+### Early Initialization Framework
+
+_This feature was championed by [Michalis Pappas](https://github.com/michpappas) with great support from [Andrei Stan](https://github.com/andreistan26),  [Oleksii Moisieiev](https://github.com/oleksiimoisieiev), [Sergiu Moga](https://github.com/mogasergiu), and [Simon Kuenzer](https://github.com/skuenzer).
+
+`libukboot` introduces `earlytab`, a lightweight framework for early initialization.
+This complements the existing `inittab`, with specific focus on the initialization of early devices and their dependencies.
+
+`libuklibparam` has been refactored to allow command-line parsing during early initialization.
+This additionally fixes a limitation of the previous registration mechanism where all parameters had to be registered from a single file.
+`libuklibparam` is assigned the highest (earliest) priority, to ensure that drivers and libraries in subsequent stages have access to Unikraft's command-line parameters. Notice that for the time being this has the side-effect of information printed by `libuklibparam` to be lost, as `earlycon` is registered to a later priority. This will be resolved once `libukprint` is introduced, as that library will provide a console buffer.
```

Comment:
```suggestion
`libuklibparam` is assigned the highest (earliest) priority, to ensure that drivers and libraries in subsequent stages have access to Unikraft's command-line parameters.
Notice that for the time being this has the side-effect of information printed by `libuklibparam` to be lost, as `earlycon` is registered to a later priority. This will be resolved once `libukprint` is introduced, as that library will provide a console buffer.
```